cnd/src/network/transport.rs

```diff
@@ -57,21 +60,24 @@ fn build_comit_transport(keypair: identity::Keypair) -> anyhow::Result<ComitTran
 /// Builds a libp2p transport with the following features:
 /// - TCP connection over the Tor network
 /// - DNS name resolution
-/// - authentication via secio
+/// - authentication via noise
 /// - multiplexing via yamux or mplex
 fn build_tor_transport(
-    keypair: identity::Keypair,
+    id_keys: identity::Keypair,
     addr: Multiaddr,
 ) -> anyhow::Result<ComitTransport> {
     let mut map = HashMap::new();
     map.insert(addr, PORT);
 
-    let transport = Socks5TokioTcpConfig::default().nodelay(true).onion_map(map);
-    let transport = DnsConfig::new(transport)?;
+    let dh_keys = noise::Keypair::<X25519Spec>::new().into_authentic(&id_keys)?;
+    let noise = NoiseConfig::xx(dh_keys).into_authenticated();
+
+    let socks = Socks5TokioTcpConfig::default().nodelay(true).onion_map(map);
+    let dns = DnsConfig::new(socks)?;
 
-    let transport = transport
+    let transport = dns
```

Comment:
Seeing this, I think DNS doesn't make a lot of sense for the Tor transport :)

Reply:
I noticed the same thing, its done so we can return the same type for both the tor transport builder function and the usual one. I elected to be lazy and not investigate further whether it was actually doing any damage since we don't use the tor one actively. Should probably be thought about more though. If you have an answer without too much brain power I'm happy to implement it.